platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -700,10 +706,16 @@ odp_ipsec_sa_t odp_ipsec_sa_create(const odp_ipsec_sa_param_t *param)
 	case ODP_AUTH_ALG_AES128_GCM:
 #endif
 	case ODP_AUTH_ALG_AES_GCM:
-		crypto_param.auth_aad_len = sizeof(ipsec_aad_t);
+		if (ipsec_sa->esn) {
+			crypto_param.auth_aad_len = 12; /* With ESN */
```

Comment:
I think the "With ESN" comments in this function are unnecessary clutter since the very previous line obviously checks if ESN is used.

Reply:
Ack, will remove this comment